packages/cmrjs/cmr-utils.js

```diff
@@ -313,41 +313,85 @@ function getS3CredentialsObject(s3CredsUrl) {
  * @param {Array<Object>} params.files - array of file objects
  * @param {string} params.distEndpoint - distribution endpoint from config
  * @param {BucketsConfig} params.buckets -  Class instance
- * @param {string} params.s3CredsEndpoint - optional s3credentials endpoint name
- * @returns {Array<{URL: string, URLDescription: string}>}
+ * @returns {Array<{URL: string, URLDescription: string, Description: string, Type: string}>}
  *   returns the list of online access url objects
  */
 function constructOnlineAccessUrls({
   files,
   distEndpoint,
-  buckets,
-  s3CredsEndpoint = 's3credentials'
+  buckets
 }) {
-  const urls = [];
-
-  files.forEach((file) => {
-    const urlObj = {};
+  const urlList = files.map((file) => {
     const bucketType = buckets.type(file.bucket);
     if (bucketType === 'protected') {
       const extension = urljoin(file.bucket, getS3KeyOfFile(file));
-      urlObj.URL = urljoin(distEndpoint, extension);
-      urlObj.URLDescription = 'File to download'; // used by ECHO10
-      urlObj.Description = 'File to download'; // used by UMMG
-      urlObj.Type = 'GET DATA'; // used by UMMG
-      urls.push(urlObj);
+      return {
+        URL: urljoin(distEndpoint, extension),
+        URLDescription: 'File to download', // used by ECHO10
+        Description: 'File to download', // used by UMMG
+        Type: 'GET DATA' // used by UMMG
+      };
     }
-    else if (bucketType === 'public') {
-      urlObj.URL = `https://${file.bucket}.s3.amazonaws.com/${getS3KeyOfFile(file)}`;
-      urlObj.URLDescription = 'File to download';
-      urlObj.Description = 'File to download';
-      urlObj.Type = 'GET DATA';
-      urls.push(urlObj);
+    if (bucketType === 'public') {
+      return {
+        URL: `https://${file.bucket}.s3.amazonaws.com/${getS3KeyOfFile(file)}`,
+        URLDescription: 'File to download',
+        Description: 'File to download',
+        Type: 'GET DATA'
+      };
     }
+    return null;
   });
 
-  urls.push(getS3CredentialsObject(urljoin(distEndpoint, s3CredsEndpoint)));
+  return urlList.filter((urlObj) => !(urlObj == null));
+}
 
-  return urls;
+/**
+ * Construct a list of ECHO10XML resource urls
+ *
+ * @param {string} distEndpoint - distribution endpoint
+ * @param {string} s3CredsEndpoint - Optional endpoint for acquiring temporary s3 creds
+ * @returns {Array<{URL: string, Description: string, Type: string}>}
+ *   returns the list of online access url objects
+ */
+function constructOnlineResourceUrls(distEndpoint, s3CredsEndpoint = 's3credentials') {
+  const credsUrl = urljoin(distEndpoint, s3CredsEndpoint);
```

Comment:
I'm pretty sure this shouldn't be joining it with the dist endpoint, it should be with the regular API endpoint. s3Credentials is off of our regular API.

Reply:
addressed in one of my most recent commits :+1:

Reply:
This is going to be changing in a pr from @flamingbear - The current implementation of 668 will continue to reference `distributionApi`

Reply:
Yup.